quality_control_oca/models/qc_test_category.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2010 NaN Projectes de Programari Lliure, S.L.
+# Copyright 2014 Serv. Tec. Avanzados - Pedro M. Baeza
+# Copyright 2014 Oihane Crucelaegui - AvanzOSC
+# Copyright 2017-2020 ForgeFlow S.L.
+# Copyright 2017 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, exceptions, fields, models
+
+
+class QcTestTemplateCategory(models.Model):
+    _name = "qc.test.category"
+    _description = "Test category"
+
+    @api.depends("name", "parent_id")
+    def _compute_get_complete_name(self):
+        for record in self:
+            names = [record.name or ""]
+            parent = record.parent_id
+            while parent:
+                names.append(parent.name)
+                parent = parent.parent_id
+            record.complete_name = " / ".join(reversed(names))
+
+    @api.constrains("parent_id")
+    def _check_recursion(self):
+        ids = self.ids
+        level = 100
+        while ids:
+            parents = self.search([("id", "in", ids), ("parent_id", "!=", False)])
+            ids = list({x.parent_id.id for x in parents})
+            if not level:
+                raise exceptions.UserError(
+                    _("Error! You can not create recursive categories.")
+                )
+            level -= 1
```

Comment:
Maybe _check_recursion ORM method can be used here. what do you think?
```suggestion
    @api.constrains("parent_id")
    def _check_parent_id(self):
        if not self._check_recursion():
            raise ValidationError(_("Error! You can not create recursive categories."))
```